Orange/widgets/visualize/owdistributions.py

```diff
@@ -91,9 +92,9 @@ class Inputs:
     settingsHandler = settings.DomainContextHandler(
         match_values=settings.DomainContextHandler.MATCH_VALUES_ALL)
     #: Selected variable index
-    variable_idx = settings.ContextSetting(-1)
+    var = settings.ContextSetting(None)
     #: Selected group variable
-    groupvar_idx = settings.ContextSetting(0)
+    group_var = settings.ContextSetting(None)
```

Comment:
Settings migrations are missing. Any saved workflows will stop workflowing.

Reply:
No, we cannot migrate context in this case. @astaric confirmed this.

Reply:
@astaric, how did this work at all, with indices as attributes? By taking the best matching domain and hoping it contains the needed attribute?!

Can we solve this by calling some migration-like method after opening the context? The domain is known by then, so we can change the index into a variable?

The alternative is to ignore it. In this case, I'd volunteer to go through all widgets and remove all uses of indices instead of variables, so we get rid of these problems asap.

Reply:
Only attributes, represented either with name or variable are considered when matching contexts. Indices require a bit of luck and usually work with a bit of code in the widget that makes sure that when an invalid index is restored from a setting, it is changed to a valid value.

Migration can be done manually in widget after open_context returns. The context can be accessed via self.current_context.

+1 for changing all widgets to store variables.

Reply:
Exactly, my question was whether we should add a patch after `openContext` to migrate the context after the domain is known, or should we say it was already rather broken before and we aren't migrating broken settings?

Reply:
I lean slightly towards making it work - converting it after openContext.

The way I see the conversion could be done is by adding a keyword_only argument `domain` to `migrate_context` on this widget and calling it after openContext.